modules/client/src/client-common/utils.ts

```diff
@@ -121,3 +146,135 @@ export function votingModeFromContracts(votingMode: number): VotingMode {
       throw new InvalidVotingModeError();
   }
 }
+
+export function applyInstallatonParamsToContract(
+  params: ApplyInstallationParams,
+): PluginSetupProcessor.ApplyInstallationParamsStruct {
+  return {
+    plugin: params.pluginAddress,
+    pluginSetupRef: {
+      pluginSetupRepo: params.pluginRepo,
+      versionTag: params.versionTag,
+    },
+    helpersHash: keccak256(
+      defaultAbiCoder.encode(["address[]"], [params.helpers]),
+    ),
+    permissions: params.permissions.map((permission) => {
+      return { ...permission, condition: permission.condition || AddressZero };
+    }),
+  };
+}
+
+export function applyInstallatonParamsFromContract(
+  result: Result,
+): DecodedApplyInstallationParams {
+  const params = result[1];
+  return {
+    helpersHash: params.helpersHash,
+    permissions: params.permissions,
+    versionTag: params.pluginSetupRef.versionTag,
+    pluginAddress: params.plugin,
+    pluginRepo: params.pluginSetupRef.pluginSetupRepo,
+  };
+}
+
+export function getNamedTypesFromMetadata(
```

Comment:
Do we have a clearer name? 
- `toStringAbi(jsonAbi)`
- `jsonAbiToNamedTypes`
- ...?

Reply:
I used the name used in the contracts, can be changed to one that makes more sense in the SDK, for me `toStringAbi` makes more sense in this context since we have functions that use the same pattern `toXXX`

So if you are ok with this one i'll prceed and change it